core/src/main/scala/quasar/plugin/postgres/datasource/PostgresDatasourceModule.scala

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright 2014–2019 SlamData Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package quasar.plugin.postgres.datasource
+
+import slamdata.Predef._
+
+import argonaut._, Argonaut._
+
+import cats.data._
+import cats.effect._
+import cats.implicits._
+
+import doobie._
+import doobie.hikari.HikariTransactor
+import doobie.implicits._
+
+import eu.timepit.refined.auto._
+
+import java.util.concurrent.Executors
+
+import org.slf4s.Logging
+
+import quasar.api.datasource.{DatasourceError => DE, DatasourceType}
+import quasar.concurrent.{BlockingContext, NamedDaemonThreadFactory}
+import quasar.connector.{LightweightDatasourceModule, MonadResourceErr}
+
+import scala.concurrent.ExecutionContext
+import scala.concurrent.duration._
+import scala.util.Random
+import scala.util.control.NonFatal
+
+import scalaz.syntax.tag._
+
+object PostgresDatasourceModule extends LightweightDatasourceModule with Logging {
+
+  type InitErr = DE.InitializationError[Json]
+
+  // The duration to await validation of the initial connection.
+  val ValidationTimeout: FiniteDuration = 10.seconds
+
+  // Maximum number of database connections per-datasource.
+  val ConnectionPoolSize: Int = 10
```

Comment:
Any reason to make this externally configurable?

Reply:
Yeah, that's probably worth an optional parameter in the config, though I'm not sure it is something we'd want to expose by default since it is easy to get wrong, from what I understand. Being able to change it without releasing a new plugin would be good though :+1: